lib/Sema/CSFix.h

```diff
@@ -621,6 +625,28 @@ class AutoClosureForwarding final : public ConstraintFix {
                                        ConstraintLocator *locator);
 };
 
+class AllowAutoClosurePointerConversion final : public ConstraintFix {
```

Comment:
Same way as diagnostic is contextual, fix could also be contextual via inheriting it from `ContextualMismatch` which takes care of storage. @sl added a new constructor there which accepts `kind` as a parameter too.